content/blog/using-the-gdb4hpc-debugger-at-scale-with-a-cuda-mpi-hpc-application.md

```diff
@@ -0,0 +1,545 @@
+---
+title: Using the gdb4hpc debugger at scale with a CUDA/MPI HPC application
+date: 2025-01-17T16:37:53.003Z
+author: Josh Chandler
+authorimage: /img/e01ld9fh0jz-u014j40c3d2-43d529500caf-192.jpg
+thumbnailimage: ""
+disable: false
+tags:
+  - HPE Cray Programming Environment
+  - gdb4hpc
+  - debuggers
+  - developer tools
+  - HPC
+  - CPE
+  - cray
+---
+Command line debuggers for Linux have existed for decades. The gdb debugger is the most famous and arguably the most powerful. But gdb has a weakness when it comes to HPC applications - it can only run on one system at a time. gdb can't be used to debug an HPC application at scale because HPC applications run tens of thousands of processes on thousands of systems at once!
+
+This is where gdb4hpc comes in. gdb4hpc is part of the HPE Cray Programming Environment package. It captures the powerful features of gdb and allows the user to apply them to an HPC application at scale. gdb4hpc can debug HPC applications running on thousands of nodes.
+
+gdb4hpc works by connecting the user to many instances of gdb at once. gdb4hpc controls all of the instances of gdb at once and aggregates and filters the results into representations that will comfortably fit on a single terminal screen.
```

Comment:
And it would really be much better to add words around it when you are referring to it at the beginning of a paragraph. Words like "tool", "debugger", "app", whatever it actually is, will help the reader have a better picture of it in their mind than just use the name of the tool. In the second sentence, you can use a pronoun, like "It controls all of the instances..."